lib/fennec/evaluator/create_permission/request.ex

```diff
@@ -46,19 +44,11 @@ defmodule Fennec.Evaluator.CreatePermission.Request do
   end
 
   defp create_permissions(params, _state, turn_state) do
-    expire_at = Fennec.Time.system_time(:second) + @lifetime
     peers = Params.get_attrs(params, Attribute.XORPeerAddress)
-    added_pemissions =
-      for peer = %Attribute.XORPeerAddress{} <- peers do
-        {peer.address, expire_at}
-      end
-
-    new_permissions =
-      added_pemissions
-      |> Enum.into(turn_state.permissions)
-
-    # Construct new protocol state and response for the client
-    new_turn_state = %TURN{turn_state | permissions: new_permissions}
+    new_turn_state = Enum.reduce(peers, turn_state,
+      fn peer, acc ->
+        TURN.put_permission(acc, peer.address)
+      end)
     new_params = %Params{params | attributes: []}
     {:respond, {new_params, new_turn_state}}
   end
```

Comment:
How about something like the following?

```Elixir
new_turn_state = peers
|> Enum.map(& Map.fetch!(&1, :address))
|> Enum.reduce(turn_state, & TURN.put_permission(&2, &1))
```

I think captures can be clearer and this way it's clear we take out the addresses to stick them into TURN structure.